test/framework/toolchain.py

```diff
@@ -339,6 +353,20 @@ def test_optarch_generic(self):
                         else:
                             self.assertFalse(generic_flags in tc.get_variable(var))
 
+    def test_optarch_aarch64_heuristic(self):
```

Comment:
@boegel This is causing some trouble I don't understand... I've added a module file for GCCcore 6.2.0 and added
```
tc = self.get_toolchain("GCCcore", version="6.2.0")
tc.prepare()
self.assertEqual(...)
```
at the end of this test.  But it is still giving me the modified `mcpu` flag based on the heuristic, even though it is definitely **not** taking the "modify option with heuristic" branch in `_set_optimal_architecture`.  Why don't I get a fresh instance with `COMPILER_OPTIMAL_ARCHITECTURE_OPTION` initialized to the defaults???